components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorService.java

```diff
@@ -0,0 +1,364 @@
+        try {
+            oAuthApplicationInfo = createDCRApplication(appName, callBackURL, grantTypes);
+            if (oAuthApplicationInfo != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Created DCR Application successfully for " + appName + ".");
+                }
+                String oAuthApplicationClientId = oAuthApplicationInfo.getClientId();
+                String oAuthApplicationCallBackURL = oAuthApplicationInfo.getCallBackURL();
+                oAuthData.addProperty(KeyManagerConstants.OAUTH_CLIENT_ID, oAuthApplicationClientId);
+                oAuthData.addProperty(KeyManagerConstants.OAUTH_CALLBACK_URIS, oAuthApplicationCallBackURL);
+                oAuthData.addProperty(KeyManagerConstants.TOKEN_SCOPES, scopes);
+                oAuthData.addProperty(KeyManagerConstants.AUTHORIZATION_ENDPOINT,
+                        storeConfigs.getAuthorizationEndpoint());
+                oAuthData.addProperty("is_sso_enabled", storeConfigs.isSsoEnabled());
+            } else {
+                String errorMsg = "No information available in OAuth application.";
+                log.error(errorMsg, ExceptionCodes.OAUTH2_APP_CREATION_FAILED);
+            }
+        } catch (APIManagementException e) {
+            String errorMsg = "Error while creating the keys for OAuth application : " + appName;
+            log.error(errorMsg, e, ExceptionCodes.OAUTH2_APP_CREATION_FAILED);
+            throw new APIManagementException(errorMsg, e, ExceptionCodes.OAUTH2_APP_CREATION_FAILED);
+        }
+        return oAuthData;
+    }
+
+    /**
+     * This method returns the access tokens for a given application.
+     *
+     * @param appName Name of the application which needs to get tokens
+     * @param requestURL Request URL with the authorization code
+     * @param grantType Grant type of the application
+     * @param userName User name of the user
+     * @param password Password of the user
+     * @param refreshToken Refresh token
+     * @param validityPeriod Validity period of tokens
+     * @return AccessTokenInfo - An object with the generated access token information
+     * @throws APIManagementException When receiving access tokens fails
+     */
+    public AccessTokenInfo getTokens(String appName, String requestURL, String grantType,
+                                     String userName, String password, String refreshToken, long validityPeriod)
+            throws APIManagementException {
+        AccessTokenInfo accessTokenInfo = new AccessTokenInfo();
+        AccessTokenRequest accessTokenRequest = new AccessTokenRequest();
+        // Get scopes of the application
+        String scopes = getApplicationScopes(appName);
+        if (log.isDebugEnabled()) {
+            log.debug("Set scopes for " + appName + " application using swagger definition.");
+        }
+        // TODO: Get Consumer Key & Secret without creating a new app, from the IS side
```

Comment:
Github issue created at,
#4299 [C5][Authenticator] Getting details of an already existing DCR application